hpcflow/sdk/helper/helper.py

```diff
@@ -255,14 +319,50 @@ def run_helper(
         timeout_check_interval = timedelta(seconds=timeout_check_interval_s)
 
     start_time = datetime.now()
+    end_time = start_time + timeout
     logger = get_helper_logger(app)
     controller = MonitorController(get_watcher_file_path(app), watch_interval, logger)
-    timeout_limit = timeout - timeout_check_interval
+    PID_vars = strip_helper_PID(app)
     try:
         while True:
-            if datetime.now() - start_time >= timeout_limit:
+            time_left_s = (end_time - datetime.now()).total_seconds()
+            if time_left_s <= 0:
                 helper_timeout(app, timeout, controller, logger)
-            time.sleep(timeout_check_interval_s)
+            time.sleep(min(timeout_check_interval_s, time_left_s))
+            # Reading args from PID file
+            PID_vars_new = strip_helper_PID(app)
+            for i in [1, 2, 3]:
```

Comment:
I think the `for` loop is a bit redundant here? If we have a new function `read_helper_args` that returns a `dict`, then these comparisons will be more explicit/readable; e.g. `if helper_args['timeout'] != new_helper_args['timeout']: ...` .

Reply:
My intention was to make the code a bit more DRY, but I realise I did not accomplish much there... I removed the for loop in [e19c5ed](https://github.com/hpcflow/hpcflow-new/pull/340/commits/e19c5edf33d6194fcb135e9f7c3f0098cc445ac9), and then reintroduced it (improved) in [bc16a75](https://github.com/hpcflow/hpcflow-new/pull/340/commits/bc16a75de36ceb084ba2936838363dc38f5b308d). I feel like it looks much better, but we can always ignore that last commit, if you prefer.

Reply:
Resolved in [e0569e4](https://github.com/hpcflow/hpcflow-new/pull/340/commits/e0569e4048b9228aea4504b4bb61e6e73569b5ef)